php/libraries/Database.class.inc

```diff
@@ -664,6 +668,10 @@ class Database
             throw new DatabaseException($err[2], $prepared->queryString, $params);
         }
 
+        if (isset($options['nofetch']) && $options['nofetch'] == true) {
```

Comment:
This might work better with !empty($options['nofetch']) in order to catch all non-truthy values. (You could also take out the isset, because empty automatically does that..)